pwiz_tools/Skyline/EditUI/UniquePeptidesDlg.cs

```diff
@@ -364,10 +364,10 @@ private void QueryPeptideProteins(ILongWaitBroker longWaitBroker)
                             var peptideDocNode = tuple.Item2;
                             HashSet<Protein> proteins = new HashSet<Protein>();
                             var peptideGroupDocNode = PeptideGroupDocNodes.First(g => ReferenceEquals(g.PeptideGroup, peptideGroup));
-                            List<Protein> proteinsForSequence;
-                            if (sequenceProteinsDict.TryGetValue(peptideDocNode.Peptide.Target.Sequence, out proteinsForSequence))
+                            // ReSharper disable once ConditionIsAlwaysTrueOrFalse

```

Comment:
Should the function above be .FirstOrDefault() instead? Otherwise, why check for null and tell ReSharper to ignore the fact that it can never be null?

Reply:
Makes sense. It was even sillier before, though - checking for null after it had already been used.

Reply:
Either way, it was a no-op before and still a no-op. So, either it should just be removed or if we think PeptideGroupDocNodes could be empty, the FirstOrDefault(). I don't claim to know whether the list can be empty or not.

Reply:
Yeah, I went with FirstOrDefault, doesn't hurt to be safety minded. If you and/or Nick can resolve the delegate thing we'll be back to green on TC code inspections.